Build fails with multiple make jobs

There is also a parallel build issue in the testsuite that I noticed. If you cannot replicate it I can attach another log. The following is from the compilation phase.

When compiling with make -j4:
make all-recursive
make[1]: Entering directory '/home/thomas/Downloads/fgsl-1.0.0'
Making all in .
make[2]: Entering directory '/home/thomas/Downloads/fgsl-1.0.0'
/bin/sh ./libtool --tag=FC --mode=compile gfortran -DHAVE_CONFIG_H -I. -I. -I. -g -O2 -c -o fgsl.lo fgsl.F90
/bin/sh ./libtool --tag=CC --mode=compile gcc -DHAVE_CONFIG_H -I. -I. -I. -g -O2 -MT libfgsl_la-fgsl_utils.lo -MD -MP -MF .deps/libfgsl_la-fgsl_utils.Tpo -c -o libfgsl_la-fgsl_utils.lo test -f 'fgsl_utils.c' || echo './'fgsl_utils.c
gfortran -DHAVE_CONFIG_H -I. -I. -I. -g -O2 -c -o fgsl.o fgsl.F90
libtool: compile: gfortran -DHAVE_CONFIG_H -I. -I. -I. -g -O2 -c fgsl.F90 -fPIC -o .libs/fgsl.o
libtool: compile: gcc -DHAVE_CONFIG_H -I. -I. -I. -g -O2 -MT libfgsl_la-fgsl_utils.lo -MD -MP -MF .deps/libfgsl_la-fgsl_utils.Tpo -c fgsl_utils.c -fPIC -DPIC -o .libs/libfgsl_la-fgsl_utils.o
libtool: compile: gcc -DHAVE_CONFIG_H -I. -I. -I. -g -O2 -MT libfgsl_la-fgsl_utils.lo -MD -MP -MF .deps/libfgsl_la-fgsl_utils.Tpo -c fgsl_utils.c -o libfgsl_la-fgsl_utils.o >/dev/null 2>&1
Fatal Error: Can't rename module file 'fgsl.mod0' to 'fgsl.mod': No such file or directory
Makefile:542: recipe for target 'fgsl.o' failed
make[2]: [fgsl.o] Error 1
make[2]: Waiting for unfinished jobs....

--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@ AM_CPPFLAGS = -I${top_srcdir} -I${top_builddir}
 
 fgslincludedir=${includedir}/fgsl
 
-all-am: fgsl.mod
+BUILT_SOURCES = fgsl.mod
 
 NULL =
 

